checkEligibility.js

```diff
@@ -36,13 +37,44 @@ function getPatronInfo (patronId) {
   })
 }
 
-function finesBlocksOrExpiration (info) {
+/**
+ *  Given an patron-info hash (i.e. one retrieved via patrons/{patronId}),
+ *  returns a Hash with the following boolean properties:
+ *   - expired: True if card expird
+ *   - blocked: True if card has blocks
+ *   - moneyOwed: True if card has > $15 fines
+ *   - ptypeDisallowsHolds: True if patron ptype disallows holds (e.g. ptype 120, 121)
+ *  Essentially, if Object.keys(hash).map((
+ */
+function identifyPatronIssues (info) {
   info = info.data.entries[0]
-  return {
+  const issues = {
     expired: new Date(info.expirationDate) < new Date(),
     blocked: info.blockInfo.code !== '-', // will want to change this once we have a list of block codes
-    moneyOwed: info.moneyOwed > 15 // may want to change this
+    moneyOwed: info.moneyOwed > 15, // may want to change this
+    ptypeDisallowsHolds: ptypeDisallowsHolds(info.patronType)
   }
+
+  // Set a single property to check for consumers that don't care *what* issues
+  // there are but only that there are issues:
+  issues.hasIssues = Object.keys(issues).filter((name) => issues[name]).length > 0
```

Comment:
Potential simplification `Object.values(issues).some(x=>x)`

Reply:
Love it. My brain is still running node v6